universum/configuration_support.py

```diff
@@ -236,51 +238,74 @@ def __eq__(self, other: Any) -> bool:
             return True
         return super().__eq__(other)
 
-    def __getitem__(self, item: str) -> Optional[str]:
+    def __getitem__(self, key: str) -> Any:
         """
         This functions simulates `dict`-like legacy read access
 
-        :param item: client-defined item
+        .. note::
+
+            It is recommended to use field-like access to non user-defined attributes of Step
+            This preserves data type information for mypy static analysis
+
+        :param key: client-defined item
         :return: client-defined value
 
-        >>> cfg = Step(name='foo', my_var='bar')
-        >>> cfg['name']
-        'foo'
-        >>> cfg['my_var']
+        >>> step = Step(name='foo', my_var='bar')
+        >>> step['name'] == step.name == 'foo'
+        True
+
+        Note that step['name'] has type 'Any', but step.name has type 'str'
+
+        >>> step['my_var']
         'bar'
-        >>> cfg['test']
+        >>> step['test']
         """
         #  _extras are checked first - just in case _extras field is added manually
         # do note that __setitem__ checks predefined fields first, however it's impossible to shadow them by
         # modifying _extras
-        return self._extras.get(item, self.__dict__.get(item, None))
+        return self._extras.get(key, self.__dict__.get(key, None))
 
     def __setitem__(self, key: str, value: Any) -> None:
         """
         This functions simulates `dict`-like legacy write access
 
+        .. note::
+
+            It is recommended to use field-like access to non user-defined attributes of Step
```

Comment:
Please add fullstops in the end of sentences; otherwise it renders a little strange in RTD (also applies to next notes too)